text/0819-public-router-service.md

```diff
@@ -0,0 +1,242 @@
+---
+Stage: Accepted
+Start Date: 2022-05-12
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js
+RFC PR: https://github.com/emberjs/rfcs/pull/819
+---
+
+<!--- 
+Directions for above: 
+
+Stage: Leave as is
+Start Date: Fill in with today's date, YYYY-MM-DD
+Release Date: Leave as is
+Release Versions: Leave as is
+Relevant Team(s): Fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies
+RFC PR: Fill this in with the URL for the Proposal RFC PR
+-->
+
+# Public `RouterService` import
+
+## Summary
+
+This RFC proposes adding a public import for [`RouterService`](https://api.emberjs.com/ember/release/classes/RouterService)
+making it extensible by addons, removing the external link assertion from `<LinkTo />` and rebuilding the `<LinkTo />` component
+to rely entirely on the public routing service instead of the internal `-router`.
+
+## Motivation
+
+Today, users of [`ember-engines`](https://github.com/ember-engines/ember-engines) are required to utilize separate
+custom components, helpers and services for linking. This leaves the experience feeling sub-par, in addition to introducing
+extra maintenance friction attempting to keep these primitives aligned with those in `ember-source`.
+
+By providing a public import for `RouterService`, `ember-engines` would be able to provide its own router service
+implementation extending from the base implementation. Combined with minor changes to the engines-specific behavior of
+`link-to`, this change would allow us to fill the existing DX gap when it comes to navigating between an Engine
+and a host App *and* allow engines to eliminate use of the legacy components required to support 4.x versions of Ember.
+
+These proposed changes would further allow `ember-engines` to deprecate the `<LinkToExternal />`
+component provided by `ember-engines` package. As a result, there will be less
+engine specific concepts and code would be more portable between regular applications and engines.
+
+Once this RFC gets implemented, following will be possible in user defined Ember Engines:
+
+* links to external routes in templates within the engine rendered via `<LinkTo />` component:
+
+  ```hbs
+  <LinkTo @route="external-route">Go Somewhere</LinkTo>
```

Comment:
I'm still thinking that we should use `<LinkTo @route="external.home">Go Somewhere</LinkTo>` where `external` (hypothetical name) means a prefix that it's an external route. It's good for ergonomics because as a developer we don't want to go to externalRoutes definition and check if it's a internal or external route. I see the importance of deprecating `<LinkToExternal/>` but I don't want to lose the readability that it means an external link without poking around the code. 

Also it solve the drawback on this RFC 😄

Reply:
@villander this would prevent addons from utilizing LinkTo, which I think makes it a non-starter. It might be interesting to bring the external-routes map into `routes.js` at some point to address this though so that there's only one location where valid routes may be defined. This would be outside the scope of this RFC.

Reply:
Looking for this perspective I agree with you @runspired I think we just need to find some way to don't make engines more confusing and harder to teach than it already is.